### ISSUE TITLE
Rename group and package modules (consistency)

### ISSUE THREAD

**Issue by rominf** [saltstack/salt#16230]
Trying to find inspirations for writing `zypper` module and inspecting `group` modules I found that the names of modules, that provide virtual modules are inconsistent. For example:

```
mac_group.py
pw_group.py
solaris_group.py
```

BUT:

```
win_groupadd.py
groupadd.py  # for *NIX
```

I propose to make define one logical naming scheme for naming _virtual_ modules.

I think that the easiest way is to name them as:

```
VIRTUALNAME_PLATFORM.py
```

Where `PLATFORM` is the OS, package manager, etc. Those values should be the same as in grains.

Examples:

```
group_osx.py
pkg_zypper.py
```

Why `pkg_zypper.py` but not `zypper_pkg.py`? Because in this case, due alphabetical sorting, all modules would be grouped together.

I think, it's not hard to do. I assume, that there is no code that relies on real module names. I found some module names occurrences in docs, but the documentation have no issues with backward compatibility.

**Comment by basepi** [comment]
I'd still love to see more standardization in the provider module naming, so I'm going to bump to keep this open.

**Comment by damon-atkins** [comment]
See https://github.com/saltstack/salt/issues/30533 `[discussion] refactor inconsistent module naming`

**Comment by thatch45** [comment]
Closed as duplicate of #30533
